### ISSUE TITLE
Constructor executed twice when overriding constructor in System.Object.

### ISSUE THREAD

**Issue by klasma** [pythonnet/pythonnet#495]
### Environment

-   Pythonnet version: 2.3.0
-   Python version: 2.7.13
-   Operating System: Windows 10

### Details

We are trying to create a subclass of System.Object (using namespace) and override the constructor.

For some reason, the conctructor is executed twice when an instance of the class is created. Below is an example that reproduces the problem.

```python
class X(System.Object):
    __namespace__ = "PyTest"

    def __init__(self):
        print("Running X constructor")

x = X()
```
The output is

```
Running X constructor
Running X constructor
```

**Comment by rickardraysearch** [comment]
Furthermore, if you add arguments to the constructor, the arguments are only passed the second time. 

```csharp
class X(System.Object):
    __namespace__ = "PyTest"

    def __init__(self, *args):
        print("Running X constructor, args = {0}".format(args))

x = X(1,2)
```
produces the output 
```
Running X constructor, args = ()
Running X constructor, args = (1, 2)
```

It seems that the first invocation is from the call in `Python.Runtime.MetaType.tp_call` to `tp_new`, whereas the second is from the call to `py__init__` using `Runtime.PyObject_Call`. 

Is the error that somehow the call to `tp_new` also calls `__init__`?

**Comment by filmor** [comment]
Hmm, the comment says

        /// Metatype __call__ implementation. This is needed to ensure correct
        /// initialization (__init__ support), because the tp_call we inherit
        /// from PyType_Type won't call __init__ for metatypes it doesn't know.

Maybe this changed in Python? I'll have a look.

**Comment by filmor** [comment]
Fixed by #503.
